src/Core/Backup/BackupGridDataProvider.php

```diff
@@ -0,0 +1,158 @@
+use PrestaShop\PrestaShop\Core\Grid\DataProvider\GridDataProviderInterface;
+use PrestaShop\PrestaShop\Core\Grid\Row\RowCollection;
+use PrestaShop\PrestaShop\Core\Grid\Search\SearchCriteriaInterface;
+use Symfony\Component\Translation\TranslatorInterface;
+
+/**
+ * Class BackupGridDataProvider provides backups for listing in grid
+ */
+final class BackupGridDataProvider implements GridDataProviderInterface
+{
+    /**
+     * @var BackupProviderInterface
+     */
+    private $backupProvider;
+
+    /**
+     * @var TranslatorInterface
+     */
+    private $translator;
+
+    /**
+     * @var string
+     */
+    private $languageDateTimeFormat;
+
+    /**
+     * @param BackupProviderInterface $backupProvider
+     * @param TranslatorInterface $translator
+     * @param string $languageDateTimeFormat
+     */
+    public function __construct(
+        BackupProviderInterface $backupProvider,
+        TranslatorInterface $translator,
+        $languageDateTimeFormat
+    ) {
+        $this->backupProvider = $backupProvider;
+        $this->translator = $translator;
+        $this->languageDateTimeFormat = $languageDateTimeFormat;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getData(SearchCriteriaInterface $searchCriteria)
+    {
+        $backups = $this->backupProvider->getBackups();
+        usort($backups, [$this, 'sortingCompare']);
+
+        $paginatedBackups = null !== $searchCriteria->getOffset() && null !== $searchCriteria->getLimit() ?
+            array_slice($backups, $searchCriteria->getOffset(), $searchCriteria->getLimit()) :
+            $backups;
+
+        $backupsArray = [];
+
+        foreach ($paginatedBackups as $backup) {
+            $backupsArray[] = [
+                'file_name' => $backup->getFileName(),
+                'file_size' => $backup->getSize(),
+                'date' => $backup->getDate(),
+                'age' => $backup->getAge(),
+                'age_formatted' => $this->getFormattedAge($backup),
+                'date_formatted' => date($this->languageDateTimeFormat, $backup->getDate()->getTimestamp()),
+                'file_size_formatted' => $this->getFormattedSize($backup),
+            ];
+        }
+
+        $backupCollection = new RowCollection($backupsArray);
+
+        return new GridData(
+            $backupCollection,
+            count($backups)
+        );
+    }
+
+    /**
+     * Compare to backup records
+     *
+     * @param BackupInterface $backup1
+     * @param BackupInterface $backup2
+     *
+     * @return bool
+     */
+    private function sortingCompare(BackupInterface $backup1, BackupInterface $backup2)
+    {
+        return $backup2->getDate()->getTimestamp() - $backup1->getDate()->getTimestamp();
+    }
+
+    /**
+     * Get formatted age
+     *
+     * @param BackupInterface $backup
+     *
+     * @return string
+     */
+    private function getFormattedAge(BackupInterface $backup)
+    {
+        if (3600 > $backup->getAge()) {
```

Comment:
For this 3600 and below 86400 values, maybe I could add class for it? For example: 

```php
if (Time::ONE_HOUR_IN_SECONDS > $backup->getAge()) {
    // do stuff
}
```

Reply:
Ok, put it in Core/Utils namespace.